webapp/components/Modal/ReportModal.vue

```diff
@@ -54,6 +91,12 @@ export default {
       const name = this.$filters.truncate(this.name, 30)
       return this.$t(`report.${this.type}.message`, { name })
     },
+    formSchema() {
+      const validReportSchema = validReport({ translate: this.$t })
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-14T19:51:50Z" title="Monday, October 14th 2019, 9:51:50 pm +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -40,10 +62,55 @@ export default {
     id: { type: String, required: true },
   },
   data() {
+    let valuesReasonCategoryOptions = [
+      'discrimination_etc',
+      'pornographic_content_links',
+      'glorific_trivia_of_cruel_inhuman_acts',
+      'doxing',
+      'intentional_intimidation_stalking_persecution',
+      'advert_products_services_commercial',
+      'criminal_behavior_violation_german_law',
+      'other',
+    ]
+
+    let reasonCategoryOptions = []
+    valuesReasonCategoryOptions.forEach(reasonCategory => {
+      reasonCategoryOptions.push({
+        label: this.$t('report.reason.category.options.' + reasonCategory),
+        value: reasonCategory,
+      })
+    })
+
     return {
       isOpen: true,
       success: false,
       loading: false,
+      failsValidations: true,
+      form: {
+        reasonCategory: null,
+        reasonCategoryOptions,
+        reasonDescription: '',
+      },
+      formSchema: {
```


I sometimes put the formSchema into `~/components/utils/` to test it separately from the component.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-15T18:02:31Z" title="Tuesday, October 15th 2019, 8:02:31 pm +02:00">Oct 15, 2019</time>_

---


done... test added... although this feels a bit overboard because the user cannot choose something that is not a predetermined options, so it must be one of the enums we provide as radio buttons
Also, we are not using the `errors`, so the disabled button is based on the `form.reasonCategory` not being `falsy`, so this is kinda acting like our `formSchema`